include/cutlass/gemm/collective/xe_mma_w8a8.hpp

```diff
@@ -0,0 +1,281 @@
+  using Copy_B = decltype(make_tiled_copy(atom_load_B{},
+                                   Layout<CopyThreadShape>{},
+                                   make_layout(shape_div(typename traits_load_B::BlockShape{}, CopyThreadShape{}))));
+  // Host side kernel arguments
+  struct Arguments {
+    ElementA  const* ptr_A;
+    StrideA dA;
+    ElementB const* ptr_B;
+    StrideB dB;
+  };
+
+  struct Params {
+    TensorMKL mA;
+    TensorNKL mB;
+  };
+
+  //
+  // Methods
+  //
+
+  CollectiveMma() = default;
+
+  template <class ProblemShape>
+  static constexpr Params
+  to_underlying_arguments(ProblemShape const& problem_shape, Arguments const& args, void* workspace) {
+    (void) workspace;
+
+    auto [M,N,K,L] = problem_shape;
+    
+    auto mA_mkl = make_tensor(make_gmem_ptr(static_cast<ElementA const*>(args.ptr_A)),
+                              make_layout(make_shape(M, K, L), args.dA));
+    auto mB_nkl = make_tensor(make_gmem_ptr(static_cast<ElementB const*>(args.ptr_B)),
+                              make_layout(make_shape(N, K, L), args.dB));
+    return Params{mA_mkl, mB_nkl};
+  }
+
+  template <class EngineIn,
+      class EngineOut,
+      class LayoutIn,
+      class LayoutOut,
+      class... Ts>
+  CUTLASS_DEVICE
+  void convert_E4M3_to_FP16(
+          Tensor<EngineIn, LayoutIn> const& tCrA_load,
+          Tensor<EngineOut, LayoutOut>& tCrA_mma) {
+
+      static_assert(is_rmem<EngineIn>::value, "Input tensor for A conversion must come from registers");
+      static_assert(is_rmem<EngineOut>::value, "Output tensor for A conversion must come from registers");
+      static_assert(cosize_v<LayoutIn> == cosize_v<LayoutOut>);
+      static_assert(size_v<LayoutIn> == cosize_v<LayoutIn>);
+      static_assert(size_v<LayoutOut> == cosize_v<LayoutOut>);
+
+      using SrcType = typename EngineIn::value_type;
+      using DstType = typename EngineOut::value_type;
+
+      static_assert(std::is_same_v<SrcType, uint8_t>, "Expected fp8 (E4M3) input as uint8_t");
+      static_assert(std::is_same_v<DstType, half_t>, "Expected fp16 output as half_t");
+
+      auto const& src = tCrA_load(_, _, _);
+      auto const& dst = tCrA_mma(_, _, _);
+
+      SrcType const* pSrc = src.data();
+      DstType* pDst = dst.data();
+
+      constexpr int num_elements = decltype(size(src))::value;
+      constexpr int vec_size = 16;
+
+      CUTLASS_PRAGMA_UNROLL
+      for (int i = 0; i < num_elements / vec_size; ++i) {
+          // load
+          cute::intel::uchar16 src_vec;
+          CUTLASS_PRAGMA_UNROLL
+          for (int j = 0; j < vec_size; ++j) {
+              src_vec[j] = pSrc[i * vec_size + j];
+          }
+          // convert fp8 -> fp16
+          cute::intel::ushort16 dst_vec = E4M3_to_FP16_vec16(src_vec);
+          // store
+          CUTLASS_PRAGMA_UNROLL
+          for (int j = 0; j < vec_size; ++j) {
+              reinterpret_cast<uint16_t*>(pDst)[i * vec_size + j] = dst_vec[j];
+
+          }
+      }
```

Comment:
Why is this loop tiled by vec_size=16? In other words: Why do you copy pSrc into a vec and call E4M3_to_FP16_vec16 rather than directly passing pSrc to E4M3_to_FP16?

Reply:
Hi @rolandschulz , thanks for review. I tested both approaches and found that the compiler couldn't vectorize  `E4M3_to_FP16`, likely because its input is a scalar. While, `E4M3_to_FP16_vec16` is explicitly vectorized, and the compiler generated SIMD loads and stores. This was verified by comparing the assembly. The scalar version (`E4M3_to_FP16`) generated **182 loads and 162 stores**, while the vectorized version (`E4M3_to_FP16_vec16`) reduced to **42 loads and 30 stores**. Ideally, the compiler should be able to vectorize `E4M3_to_FP16` and have the same behavior as `E4M3_to_FP16_vec16`. Currently, it doesn't do so, and seems its a problem of the compiler. 

As for the vector size of 16, it is because the current SIMD lane size is 16.

Reply:
@min-jean-cho Good analysis! I think this is the problem of compiler so we have to do vectorization manually. @rolandschulz are you used to file JIRA for these kind of optimizations?

Reply:
> As for the vector size of 16, it is because the current SIMD lane size is 16

As communicated over email, all elements in the array of a specific work-item correspond to the same SIMD lane on an EU, so vector size shouldn't matter.

![image](https://github.com/user-attachments/assets/bf302255-ea0f-40ef-995f-6c59ac199f2d)